src/Simulator/IncrementalRedstoneSimulator.cpp

```diff
@@ -185,24 +169,24 @@ void cIncrementalRedstoneSimulator::SimulateChunk(std::chrono::milliseconds a_Dt
 
 	for (auto dataitr = m_RedstoneSimulatorChunkData->m_ChunkData.begin(); dataitr != m_RedstoneSimulatorChunkData->m_ChunkData.end();)
 	{
-		if (dataitr->DataTwo)
+		if (dataitr->second.second)
 		{
 			dataitr = m_RedstoneSimulatorChunkData->m_ChunkData.erase(dataitr);
 			continue;
 		}
 
-		switch (dataitr->Data)
+		switch (dataitr->second.first)
```

Comment:
This huge switch should really be replaced with a lookup table, but I suspect that's out of scope for this PR.